notebooks/3.jump-analysis/nbconverted/3.jump_analysis.py

```diff
@@ -0,0 +1,190 @@
+print("Number of Metadata Features:", len(meta_features))
+print(
+    "Number of CP features that cell injury has",
+    len(cell_injury_cp_feature_cols["feature_cols"]),
+)
+print("Number of CP features that JUMP has:", len(jump_cp_features))
+
+
+# Now that we have identified the features present in both datasets, the next step is to align them. This involves identifying the common features between both profiles and utilizing these features to update our JUMP dataset for our machine learning model.
+
+# In[5]:
+
+
+cell_injury_cp_features = cell_injury_cp_feature_cols["feature_cols"]
+
+# finding shared featues using intersection
+aligned_features = list(set(cell_injury_cp_features) & set(jump_cp_features))
+
+# displaying number of shared features between both profiles
+print("Number of shapred features of both profiles", len(aligned_features))
+
+
+# The objective of this step is to preserve the order of the feature space.
+#
+# Since we have identified the shared feature space across both profiles, we still need to address those that are missing.
+# Therefore, to maintain the feature space order, we opted to use the cell injury feature space as our reference feature space order, as our multi-class model was trained to understand this specific order.
+#
+# Next, we addressed features that were not found within the JUMP dataset.
+# This was done by including them in the alignment process, but defaulted their values to 0.
+#
+# Ultimately, we generated a new profile called `aligned_jump_df`, which contains the correctly aligned and ordered feature space from the cell injury dataset.
+
+# In[6]:
+
+
+# multiplier is the number of samples in JUMP data
+# this is used to default non-aligned features to 0
+multiplier = jump_df.shape[0]
+
+# storing feature and values in order
+aligned_jump = {}
+for injury_feat in cell_injury_cp_features:
+    if injury_feat not in aligned_features:
+        aligned_jump[injury_feat] = [0.0] * multiplier
+    else:
+        aligned_jump[injury_feat] = jump_df[injury_feat].values.tolist()
+
+# creating dataframe with the aligned features and retained feature order
+aligned_jump_feats_df = pd.DataFrame.from_dict(aligned_jump, orient="columns")
+
+# sanity check: see if the feature order in the `cell_injury_cp_feature_cols` is the same with
+# the newly generated aligned JUMP dataset
+assert (
+    cell_injury_cp_features == aligned_jump_feats_df.columns.tolist()
+), "feature space are not aligned"
+assert check_feature_order(
+    ref_feat_order=cell_injury_cp_features,
+    input_feat_order=aligned_jump_feats_df.columns.tolist(),
+), "feature space do not follow the same order"
+
+
+# ## Applying to our Multi-Class trained model
+#
+# We applying the aligned JUMP dataset to our trained multi-class model and measure the probabiltiies of which cell injury each well possessed.
+
+# In[7]:
+
+
+# loading in mutliclass model
+multi_class_cell_injury_model = joblib.load(multi_class_model_path)
```

Comment:
You might consider also adding the shuffled model here (alternatively, you can do this in a later PR, which might also make sense)

Reply:
Created an issue! Will work on this later!